src/oop/special_class_method/medium/q2/MainTest.java

```diff
@@ -0,0 +1,198 @@
+        return Stream.of(
+                0.0, 23.3233, 553233333.55655334324, -111111.333342, 3554 / 3.7
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("doubleInputProvider")
+    public void correctGetBrightnessMethod(double value) throws Throwable {
+        Object[][] arguments = {
+                {value, double.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        assertTrue(testClass.hasMethod(getAttributeMethodName1, null, double.class, new String[]{"public"}),
+                "Your " + className + " class does not include the correct " + getAttributeMethodName1 + " method.");
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName1, testInstance);
+        assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"),
+                "Your " + getAttributeMethodName1 + " method does not have the correct visibility modifier.");
+        _assertEquals(value, getMethodOutput,
+                "Your " + getAttributeMethodName1 + " method does not return the value of the " + attributeName1 + " field.");
+    }
+
+    private static Stream<Arguments> twoDoubleInputProvider() {
+        return Stream.of(
+                Arguments.of(5.4, 6.4),
+                Arguments.of(654.3, 54.2),
+                Arguments.of(43.3, 43.2),
+                Arguments.of(432.2, 899.8)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoDoubleInputProvider")
+    public void correctSetBrightnessMethod(double initialValue, double updatedValue) throws Throwable {
+        Object[][] instantiationArguments = {
+                {initialValue, double.class}
+        };
+        Object testInstance = testClass.createInstance(instantiationArguments);
+        Object[][] setMethodArguments = {
+                {updatedValue, double.class}
+        };
+        Class<?>[] methodModifierClasses = {
+                double.class
+        };
+        assertTrue(testClass.hasModifier(setAttributeMethodName1, methodModifierClasses, "public"),
+                "Your " + setAttributeMethodName1 + " method does not have the correct visibility modifier.");
+        assertTrue(testClass.hasMethod(setAttributeMethodName1, methodModifierClasses, Void.TYPE, new String[]{"public"}),
+                "Your " + className + " class does not include the correct " + setAttributeMethodName1 + " method.");
+        String incorrectSetterMessage =
+                "Your " + setAttributeMethodName1 + " method does not correctly update the value of " + attributeName1 + ".";
+        Object setMethodOutput = testClass.callMethod(setAttributeMethodName1, setMethodArguments, testInstance);
+        _assertEquals(updatedValue, testClass.getFieldValue(testInstance, attributeName1),
+                incorrectSetterMessage);
+        assertNull(setMethodOutput, String.join(" ", "Your", setAttributeMethodName1, "method should not return any output"));
```

Comment:
this message should get moved up to the hasMethod assert, and then this assert gotten rid of since Void.TYPE ensures no return value